### PR TITLE
DX-341 add example recipe for glob matching

### PR DESCRIPTION
**Pull request by mccataldo**
Closes DX-341

### CODE REVIEW

**Comment by mccataldo** [comment]
@jennifer-shehane  is there something I need to do to fix this or is it something you need to fix for me?

> [CLA assistant check](https://cla-assistant.io/cypress-io/cypress-example-recipes?pullRequest=723)
> Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/cypress-io/cypress-example-recipes?pullRequest=723) before we can accept your contribution.
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/cypress-io/cypress-example-recipes?pullRequest=723) it.

**Comment by jennifer-shehane** [comment]
@mccataldo yah sometimes the repo just needs to be rechecked. It’s just buggy. 

Can you try to visit https://cla-assistant.io/ (I think it just follows same permissions as GH) and click ‘Recheck PRs’ under the `cypress-documentation` repo. That usually resolves it instantly. If you don’t have access maybe I have to do it.

File: examples/stubbing-spying__intercept/cypress/integration/glob-matching-intercept-url.spec.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+describe('intercept', () => {
+    cy.intercept('/api/roles')
```
Comment (cowboy) [inline examples/stubbing-spying__intercept/cypress/integration/glob-matching-intercept-url.spec.js:2]: Can we indent this file with 2 spaces? That seems to be the standard in this repo (but I may have missed it!)

File: examples/stubbing-spying__intercept/cypress/integration/glob-matching-intercept-url.spec.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    // will match https://example.com/api/roles
+    // will not match https://example.com/api/roles?country=US
+    
+    cy.intercept('/api/roles*')
+    // will match https://example.com/api/roles
+    // will match https://example.com/api/roles?country=US
+    
+    cy.intercept('/api/roles?*utm_source=+(My%App|https*)')
+    // will match https://example.com/api/roles?utm_medium=Nav&utm_campaign=Docs&utm_source=My%App
+    // will match https://example.com/api/roles?utm_medium=Nav&utm_campaign=Docs&utm_source=https://....
```
Comment (cowboy) [inline examples/stubbing-spying__intercept/cypress/integration/glob-matching-intercept-url.spec.js:12]: Can you add an example that doesn't match here?

File: examples/stubbing-spying__intercept/cypress/integration/glob-matching-intercept-url.spec.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+describe('intercept', () => {
+    cy.intercept('/api/roles')
+    // will match https://example.com/api/roles
+    // will not match https://example.com/api/roles?country=US
+    
+    cy.intercept('/api/roles*')
+    // will match https://example.com/api/roles
+    // will match https://example.com/api/roles?country=US
+    
+    cy.intercept('/api/roles?*utm_source=+(My%App|https*)')
```
Comment (cowboy) [inline examples/stubbing-spying__intercept/cypress/integration/glob-matching-intercept-url.spec.js:10]: Is `My%App` correct? I'd expect to see either `My App` or `My%20App` here (and in the matched URLs). Perhaps `MyApp` would be better.

**Review by cowboy — CHANGES_REQUESTED** [review]

File: examples/stubbing-spying__intercept/cypress/integration/glob-matching-intercept-url.spec.js (hunk 1 above)
Comment (mccataldo) [reply examples/stubbing-spying__intercept/cypress/integration/glob-matching-intercept-url.spec.js:2]: VSC was formatting with Prettier defaults. I fixed all broken eslint rules.

File: examples/stubbing-spying__intercept/cypress/integration/glob-matching-intercept-url.spec.js (hunk 2 above)
Comment (mccataldo) [reply examples/stubbing-spying__intercept/cypress/integration/glob-matching-intercept-url.spec.js:12]: Done

File: examples/stubbing-spying__intercept/cypress/integration/glob-matching-intercept-url.spec.js (hunk 3 above)
Comment (mccataldo) [reply examples/stubbing-spying__intercept/cypress/integration/glob-matching-intercept-url.spec.js]: Done

File: examples/stubbing-spying__intercept/cypress/integration/glob-matching-intercept-url.spec.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  cy.intercept('/api/roles')
+  // will match https://example.com/api/roles
+  // will not match https://example.com/api/roles?country=US
+
+  cy.intercept('/api/roles*')
+  // will match https://example.com/api/roles
+  // will match https://example.com/api/roles?country=US
+
+  cy.intercept('/api/roles?*utm_source=+(MyApp|https*)')
+  // will match https://example.com/api/roles?utm_medium=Nav&utm_campaign=Docs&utm_source=MyApp
```
Comment (cowboy) [inline examples/stubbing-spying__intercept/cypress/integration/glob-matching-intercept-url.spec.js:11]: Could you remove either the `utm_medium` or `utm_campaign` parameter from the examples just to make the lines a little shorter and easier to read?

**Review by cowboy — CHANGES_REQUESTED** [review]

File: examples/stubbing-spying__intercept/cypress/integration/glob-matching-intercept-url.spec.js (hunk 4 above)
Comment (mccataldo) [reply examples/stubbing-spying__intercept/cypress/integration/glob-matching-intercept-url.spec.js]: Done

**Review by cowboy — APPROVED** [review]

**Review by cowboy — APPROVED** [review]

**Review by cowboy — APPROVED** [review]

**Review by cowboy — APPROVED** [review]

**Pull request merged**